lib/cocoapods-core/cdn_source.rb

```diff
@@ -118,33 +142,41 @@ def versions(name)
 
       return nil if @version_arrays_by_fragment_by_name[fragment][name].nil?
 
-      concurrent_requests_catching_errors do
-        loaders = []
+      errors = []
 
-        @versions_by_name[name] ||= @version_arrays_by_fragment_by_name[fragment][name].map do |version|
+      Sync do
+        @version_arrays_by_fragment_by_name[fragment][name].each do |version|
           # Optimization: ensure all the podspec files at least exist. The correct one will get refreshed
           # in #specification_path regardless.
           podspec_version_path_relative = Pathname.new(version).join("#{name}.podspec.json")
 
           unless pod_path_actual.join(podspec_version_path_relative).exist?
             # Queue all podspec download tasks first
-            loaders << download_file_async(pod_path_relative.join(podspec_version_path_relative).to_s)
-          end
-
-          begin
-            Version.new(version) if version[0, 1] != '.'
-          rescue ArgumentError
-            raise Informative, 'An unexpected version directory ' \
-            "`#{version}` was encountered for the " \
-            "`#{pod_path_actual}` Pod in the `#{name}` repository."
+            Async::Task.current.async do
+              download_file_async(pod_path_relative.join(podspec_version_path_relative).to_s)
+            rescue ::StandardError => e
+              errors << [version, e.message]
+            end
           end
-        end.compact.sort.reverse
+        end
+      ensure
+        http_client.close
```

Comment:
You don't need to do this any more. It's implicit when the reactor goes out of scope.

Reply:
@ioquatix I've tried removing it and it has caused odd timeouts when I tried doing it in my production project.

Reply:
Probably because you are using `retries: 0`. You definitely shouldn't need this, and to take full advantage of persistent connections you should not be doing this.

Reply:
I was suspect async-http builtin retry would break CocoaPods behavior (give user a warning when request error), just trying to remove `retries: 0`, tests still pass, so I remove `retries: 0`